service/src/main/java/skills/services/SelfReportingService.groovy

```diff
@@ -117,14 +123,22 @@ class SelfReportingService {
         List<UserRoleRes> userRoleRes = accessSettingsStorageService.getUserRolesForProjectId(skillDefinition.projectId)
                 .findAll { it.roleName == RoleName.ROLE_PROJECT_ADMIN }
         ProjDef projDef = projDefRepo.findByProjectId(skillDefinition.projectId)
-        Notifier.NotificationRequest request = new ApprovalNotificationRequestBuilder(
-                userRequesting: userId,
-                requestMsg: requestMsg,
+        UserAttrs userAttrs = userAttrsRepo.findByUserId(userId)
+        Notifier.NotificationRequest request = new Notifier.NotificationRequest(
                 userIds: userRoleRes.collect { it.userId },
-                skillDef: skillDefinition,
-                publicUrl: publicUrl,
-                projectName: projDef.name
-        ).build()
+                type: Notification.Type.SkillApprovalRequested.toString(),
+                keyValParams: [
+                        userRequesting: userAttrs.userIdForDisplay,
+                        numPoints     : skillDefinition.pointIncrement,
+                        skillName     : skillDefinition.name,
+                        approveUrl    : "${publicUrl}administrator/projects/${skillDefinition.projectId}/self-report",
```

Comment:
do we guarantee anywhere that publicUrl ends with a trailing slash?

Reply:
@mosefer yes in ``skills.services.SelfReportingService#getPublicUrl``